core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/returnfirstnonnull/ReturnFirstNonNullUnitTest.java

```diff
@@ -0,0 +1,108 @@
+package returnfirstnonull;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.lang3.ObjectUtils;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Predicates;
+import com.google.common.collect.Iterables;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ReturnFirstNonNullUnitTest {
+
+    private List<Object> objects;
+
+    @Before
+    public void init() {
+        objects = Arrays.asList(
+            null,
+            new Object(),
+            new Object()
+        );
+    }
+
+    @Test
+    public void givenListOfObjects_whenIteratingWithForLoop_thenReturnFirstNonNull() {
```

Comment:
It seems a bit overkill to have a test for both an indexed for loop and an enhanced for loop. I would delete one of them unless you have a strong reason to keep both?

Reply:
Agree, removed enhanced for loop